it/controllers/pensions/incomeFromPensions/RemovePensionSchemeControllerISpec.scala

```diff
@@ -0,0 +1,211 @@
+  }
+
+  object CommonExpectedEN extends CommonExpectedResults {
+    val expectedTitle = "Are you sure you want to remove pension name 1?"
+    val expectedHeading = "Are you sure you want to remove pension name 1?"
+    val expectedCaption: Int => String = (taxYear: Int) => s"Income from pensions for 6 April ${taxYear - 1} to 5 April $taxYear"
+    val buttonText = "Remove pension"
+    val cancelText = "Cancel"
+  }
+
+  object CommonExpectedCY extends CommonExpectedResults {
+    val expectedTitle = "Are you sure you want to remove pension name 1?"
+    val expectedHeading = "Are you sure you want to remove pension name 1?"
+    val expectedCaption: Int => String = (taxYear: Int) => s"Income from pensions for 6 April ${taxYear - 1} to 5 April $taxYear"
+    val buttonText = "Remove pension"
+    val cancelText = "Cancel"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, String]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, None),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, None)
+  )
+
+  ".show" when {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)}" should {
+
+        "render the Remove Pension Scheme page" which {
+
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            insertCyaData(aPensionsUserData, aUserRequest)
+            urlGet(fullUrl(removePensionSchemeUrl(taxYearEOY, Some(0))), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(expectedTitle)
+          h1Check(expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          buttonCheck(buttonText)
+          linkCheck(cancelText, cancelLinkSelector, s"${ukPensionSchemeSummaryListUrl(taxYearEOY)}")
+        }
+      }
+    }
+
+    "no data is returned" should {
+
+      "redirect to the Pensions Summary page" should {
+
+        lazy val result: WSResponse = {
+          dropPensionsDB()
+          authoriseAgentOrIndividual(isAgent = false)
+          urlGet(fullUrl(removePensionSchemeUrl(taxYearEOY, Some(0))), follow = false,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+        }
+
+        s"has a SEE_OTHER ($SEE_OTHER) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location").contains(pensionSummaryUrl(taxYearEOY)) shouldBe true
+        }
+      }
+    }
+
+    "redirect the user to the UK Pension Income Summary page" when {
+
+      "there is no pensionSchemeIndex" should {
```

Comment:
think this test and below are switched? This one has an invalid index, and the below one has no index